n3fit/src/evolven3fit/evolve.py

```diff
@@ -0,0 +1,204 @@
+        eko_op = output.Output.load_tar(eko_path)
+    else:
+        eko_op = eko_utils.construct_eko_for_fit(theory, op, log, dump_eko)
+    eko_op.xgrid_reshape(targetgrid=x_grid, inputgrid=x_grid)
+    info = gen_info.create_info_file(theory, op, 1, info_update={})
+    info["NumMembers"] = "REPLACE_NREP"
+    info["ErrorType"] = "replicas"
+    info["AlphaS_Qs"] = list(eko_op["Q2grid"].keys())
+    info["XMin"] = float(x_grid[0])
+    info["XMax"] = float(x_grid[-1])
+    dump_info_file(usr_path, info)
+    for replica in initial_PDFs_dict.keys():
+        evolved_block = evolve_exportgrid(initial_PDFs_dict[replica], eko_op, x_grid)
+        dump_evolved_replica(
+            evolved_block, usr_path, int(replica.removeprefix("replica_"))
+        )
+    # remove folder:
+    # The function dump_evolved_replica dumps the replica files in a temporary folder
+    # We need then to remove it after fixing the position of those replica files
+    (usr_path / "nnfit" / usr_path.stem).rmdir()
+
+
+def load_fit(usr_path):
+    """
+    Loads all the replica pdfs at fitting scale in usr_path and return the exportgrids
+
+    Parameters
+    ----------
+
+        usr_path: pathlib.Path
+            path to the folder containing the fit
+
+    Returns
+    -------
+
+            : dict
+            exportgrids info
+    """
+    nnfitpath = usr_path / "nnfit"
+    replica_list = []
+    for yaml_file in nnfitpath.glob("replica_*/*.exportgrid"):
+        replica_list.append(yaml_file)
+    pdf_dict = {}
+    for replica in replica_list:
+        data = yaml.safe_load(replica.read_text())
+        pdf_dict[replica.parent.stem] = data
+    return pdf_dict
+
+
+def evolve_exportgrid(exportgrid, eko, x_grid):
+    """
+    Evolves the provided exportgrid for the desired replica with the eko and returns the evolved block
+
+    Parameters
+    ----------
+        exportgrid: dict
+            exportgrid of pdf at fitting scale
+        eko: eko object
+            eko operator for evolution
+        xgrid: list
+            xgrid to be used as the targetgrid
+    Returns
+    -------
+        : np.array
+        evolved block
+    """
+    # construct LhapdfLike object
+    pdf_grid = np.array(exportgrid["pdfgrid"]).transpose()
+    pdf_to_evolve = utils.LhapdfLike(pdf_grid, exportgrid["q20"], x_grid)
+    # evolve pdf
+    evolved_pdf = apply.apply_pdf(eko, pdf_to_evolve)
+    # generate block to dump
+    targetgrid = list(eko["targetgrid"])
+    def ev_pdf(pid, x, Q2):
+        return x * evolved_pdf[Q2]["pdfs"][pid][targetgrid.index(x)]
+    block = genpdf.generate_block(
+        ev_pdf,
+        xgrid=targetgrid,
+        Q2grid=list(eko["Q2grid"].keys()),
+        pids=br.flavor_basis_pids,
+    )
+    return block
+
+
+def dump_evolved_replica(evolved_block, usr_path, replica_num):
+    """
+    Dump the evolved replica given by evolved_block as the replica num "replica_num" in
+    the folder usr_path/nnfit/usr_path.stem.
```

Comment:
```suggestion
    the folder usr_path/nnfit/replica_num/usr_path.dat
```
This would be the right output place right?

Reply:
Yes it is. However this would not work with `genpdf.export.dump_blocks`. In fact, as you can see here
https://github.com/N3PDF/eko/blob/01efee1d27d215e895dd055debff8905af3d548c/src/ekobox/genpdf/export.py#L65
this function creates a folder called `path_where_dump.stem` by default. This is the reason why we have this temporary folder and then we need to move the files. I guess this answers also your other question. The point is that if `path_where_dump = nnfit/replica_{replica_num}`, then genpdf would create a folder called `replica_{replica_num}` for each replica (and that would be worst).

Reply:
No, my point is that with the changes the docstr in `dump_evolved_replica` would be wrong, since now things are effectively dumped to the right place.

On the other hand, your comment answers my question somewhere else about the eko where to dump stuff.

Reply:
Ah yes you are right